SR2021/2021-04-20-league-4-knockouts-simulator-update.md

```diff
@@ -0,0 +1,58 @@
+---
+to: All SR2021 teams
+subject: Student Robotics League 4, Finals & Simulator Update
+---
+
+Hi,
+
+With the final round of league matches coming up this weekend and the finals the
+week after, we wanted to give you some quick reminders about the schedule. We've
+also fixed a small bug in the compass which should make it more accurate.
+
+## Simulator Update
+
+We've released a new version of the simulator, which includes a fix to the way
+that noise is treated within the compass. Previously the noise wasn't quite
+random and would tend to be biased towards one side. The noise is now evenly
+balanced and we have slightly reduced the bounds of the noise to reflect real
+hardware.
```

Comment:
The main change here is fixing the bug that higher compass values have increased noise, the biasing to one side is very much an edge-case which was exaggerated by the other issue.

Reply:
Yeah, this isn't a perfect description. I'm trying to convey a high level understanding of the bug without getting too technical.
Do you have a suggestion for an improvement here?

Reply:
Ultimately the bias issue really needs mentioning.
```suggestion
that noise is treated within the compass. Previously certain circumstances caused
excessive noise to be added to the compass. The noise is now a consistent
level and we have slightly reduced the bounds of the noise to reflect real
hardware.
```

Reply:
I'm not sure that removing the mention of the bias adds is beneficial here though?

Consider the case of a team which has applied a noise cancellation technique here (perhaps multiple querying). If we don't mention the bias they could well believe that they don't need to upgrade for their code to work. They could then end up trying to diagnose why their robot has a subtle turn one way when using the compass and not realise that this could be the cause. Worse they could put in a compensation fudge factor which would then disadvantage them in the leagues.

Reply:
```suggestion
that noise is treated within the compass. Previously certain circumstances caused
excessive noise to be added to the compass. There were also situations where this
noise would be biased towards one side. The noise is now a consistent, balanced level
and we have slightly reduced the level of noise to reflect real hardware.
```

Reply:
0e7eaa562cd193a9b48283a75216e1ac6ec02c5c